### PR TITLE
Show all catalog entires

### PR DESCRIPTION
**Pull request by Himangini**
### Description
Added --all option for search cmd

### Checklist
- [x] Added tests that cover your change
- [x] Added/modified documentation as required (such as the `README.md`)
- [x] Made sure the title of the PR is a good description that can go into the release notes
- [x] Added a `kind` label to the PR (e.g. `kind/feature`)

### CODE REVIEW

File: cmd/pctl/main.go [hunk 1]
```diff
@@ -93,6 +93,21 @@ func parseArgs(c *cli.Context) (string, *client.Client, error) {
...
+		ServiceName:    c.String("catalog-service-name"),
+		ServicePort:    c.String("catalog-service-port"),
+	}
+
+	client, err := client.NewFromOptions(options)
+	if err != nil {
+		return nil, err
+	}
+	return client, nil
+}
```
Comment (Callisto13) [inline cmd/pctl/main.go:109]: can we shorten this to 
```suggestion
	return client.NewFromOptions(options)
}
```

File: cmd/pctl/search.go [hunk 2]
```diff
@@ -25,41 +25,43 @@ func searchCmd() *cli.Command {
...
-				_ = cli.ShowCommandHelp(c, "search")
-				return err
-			}
-
-			profiles, err := catalog.Search(catalogClient, searchName)
-			if err != nil {
-				return err
-			}
+			var profiles []profilesv1.ProfileCatalogEntry
 			outFormat := c.String("output")
```
Comment (Callisto13) [inline cmd/pctl/search.go:36]: can we move this closer to where it is used?

File: cmd/pctl/search.go [hunk 3]
```diff
@@ -25,41 +25,43 @@ func searchCmd() *cli.Command {
...
-			out, err := f.Format(getter)
-			if err != nil {
-				return err
+			outErr := outputFormatter(profiles, outFormat)
+			if outErr != nil {
+				return outErr
 			}
 
-			fmt.Println(out)
 			return nil
```
Comment (Callisto13) [inline cmd/pctl/search.go:65]: I think we can just `return outputFormatter(profiles, outFormat)` here too.

Can we rename this func too? With `outputFormatter` I feel like I should be getting an object back. What do you think of just `formatOutput` or `printOutput`?

File: cmd/pctl/search.go (hunk 2 above)
Comment (Himangini) [reply cmd/pctl/search.go]: its used in the following if-else loop. i have moved the `outFormat` var down the line where its used

File: cmd/pctl/main.go (hunk 1 above)
Comment (Himangini) [reply cmd/pctl/main.go]: done 👍🏻

File: cmd/pctl/search.go (hunk 3 above)
Comment (Himangini) [reply cmd/pctl/search.go]: done , renamed it to `formatOutput` 👍🏻

File: cmd/pctl/search.go [hunk 4]
```diff
@@ -25,42 +25,38 @@ func searchCmd() *cli.Command {
...
+				if err != nil {
+					_ = cli.ShowCommandHelp(c, "search")
+					return err
+				}
+				profiles, err = catalog.Search(catalogClient, "all")
+				if err != nil {
+					return err
+				}
+			} else {
+				searchName, catalogClient, err := parseArgs(c)
```
Comment (aclevername) [inline cmd/pctl/search.go:47]: should we error if `--all` and an arg is provided?

File: pkg/catalog/search.go [hunk 5]
```diff
@@ -9,11 +9,21 @@ import (
...
 func Search(catalogClient CatalogClient, searchName string) ([]profilesv1.ProfileCatalogEntry, error) {
-	q := map[string]string{
-		"name": searchName,
+	var data []byte
+	var statusCode int
+	var err error
+
+	if searchName == "all" {
+		data, statusCode, err = catalogClient.DoRequest("/profiles", nil)
+	} else {
```
Comment (aclevername) [inline pkg/catalog/search.go:20]: does this mean `pctl search all` is equivalent to `pctl search --all`? Also it blocks users from searching for the string `all`, which might be fine if its intentional :smile:

File: cmd/pctl/search.go (hunk 4 above)
Comment (Himangini) [reply cmd/pctl/search.go:47]: i am not sure , what do u think? atm i think if u specify `-all` and searchname it'll ignore the searchname and return all available profiles

File: pkg/catalog/search.go (hunk 5 above)
Comment (Himangini) [reply pkg/catalog/search.go:20]: i think this was not intentional, i'm gonna remove this. nice catch 👌🏻

File: cmd/pctl/search.go (hunk 4 above)
Comment (aclevername) [reply cmd/pctl/search.go:47]: I think its worth erroring if both are set, to avoid confusion. People might think `--all` does something else otherwise

File: Makefile [hunk 6]
```diff
@@ -26,7 +26,7 @@ local-env: submodule ## Create local environment
 
 submodule: ## Update git submodules
 	git submodule init
-	git submodule update
+	git submodule update --remote --merge
```
Comment (Skarlso) [inline Makefile:29]: This might mess things up when there is a zero checkout. The init and update call are fine if there is no module initially, which is the case on CI when this is running. If you want to update, that's when we force a remote and a merge.

Comment (Himangini) [reply Makefile:29]: i had to do this to get the local tests working , i am gonna put it back to what it was before

Comment (Skarlso) [reply Makefile:29]: Hmm. That is interesting. Maybe leave it in and we'll see?

Comment (Skarlso) [reply Makefile:29]: Oh! I remember why it's that. Because this might cause an involuntary update to running tests locally. Normally, you don't want to force update.

Comment (Himangini) [reply Makefile]: i have removed it from the makefile still getting same lint error 😑

File: cmd/pctl/search.go (hunk 4 above)
Comment (Himangini) [reply cmd/pctl/search.go]: i have added the error if both `-all` and searchname is provided

File: cmd/pctl/main.go [hunk 7]
```diff
@@ -93,6 +93,21 @@ func parseArgs(c *cli.Context) (string, *client.Client, error) {
...
+		ServiceName:    c.String("catalog-service-name"),
+		ServicePort:    c.String("catalog-service-port"),
+	}
+
+	if c.Args().Len() > 0 {
+		return nil, fmt.Errorf("argument must not be provided")
+	}
+
+	return client.NewFromOptions(options)
+}
```
Comment (aclevername) [inline cmd/pctl/main.go:109]: is this a repeat of the above? It could be re-used instead of duplicated :smile:

**Review by aclevername — APPROVED** [review]
1 comment otherwise LGTM :smile:

**Comment by Skarlso** [comment]
I have no idea why the linter is broken. :O

File: cmd/pctl/main.go (hunk 7 above)
Comment (Himangini) [reply cmd/pctl/main.go:109]: yeah I was thinking of refactoring it but got sidetracked by the failing build 😓  I'll refactor it 👍🏻

Comment (Himangini) [reply cmd/pctl/main.go]: done 👍🏻

File: cmd/pctl/search.go [hunk 8]
```diff
@@ -25,42 +25,38 @@ func searchCmd() *cli.Command {
...
+				if err != nil {
+					_ = cli.ShowCommandHelp(c, "search")
+					return err
+				}
+				profiles, err = catalog.Search(catalogClient, searchName)
+				if err != nil {
+					return err
 				}
 			}
```
Comment (Callisto13) [inline cmd/pctl/search.go:56]: could this be simplified and dried a bit? (not tested, btw, could be wrong)
```suggestion
      searchName, catalogClient, err := parseArgs(c)
      if err != nil {
        _ = cli.ShowCommandHelp(c, "search")
        return err
      }
      if c.Bool("all") {
        searchName = ""
      }
      profiles, err = catalog.Search(catalogClient, searchName)
      if err != nil {
        return err
      }
```

**Review by Callisto13 — APPROVED** [review]
lgtm 👍 

could you rebase your commit messages before you merge so that their context is clear?
